### PR TITLE
Remove comparator on mobile fair collections

### PR DESCRIPTION
**Pull request by williardx**
Fixes issue where fair dates are occasionally displayed incorrectly on mobile but not on desktop. The root issue is a comparator on the mobile fairs collection that sorts fairs based on banner image size, overriding any other sort order passed to `/fairs`. Incidentally fixes a related issue where fairs in the "Previous Fairs" tab on mobile might show in a jumbled order.

- Removes comparator in mobile fairs collection that sorts 
- Explicitly sorts fairs in chronological order based on `start_at`
- Formats fair dates using `formatDates` instead of `subtitle`
- Includes update to `yarn.lock` setting CoffeeScript version

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please assign someone to your PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: src/mobile/collections/fairs.coffee [hunk 1]
```diff
@@ -12,15 +12,6 @@ module.exports = class Fairs extends Backbone.Collection
...
   model: Fair
 
-  comparator: (fair) ->
-    sizes =
-      'x-large' : 1
-      'large' : 2
-      'medium' : 3
-      'small' : 4
-      'x-small' : 5
-    sizes[fair.get('banner_size')]
```
Comment (starsirius) [inline src/mobile/collections/fairs.coffee]: I did a brief search in Force under mobile and looked like there is no other places do something like `fairs.fetch`. Modifying the default comparator here might be fine, although this sorting order makes some sense to me (sorting by relative size.) A safer alternative to keep the default sorting for other use cases is to [override](http://backbonejs.org/#Collection-comparator) it just for the fairs we fetch from the fair organizer app. We can do something like
```coffeescript
fair = new Fairs
fair.comparator = #...
```

File: src/mobile/apps/fair_organizer/routes.coffee [hunk 2]
```diff
@@ -42,6 +42,7 @@ module.exports.fetchFairOrgData = (req, res, next) ->
   request = fairs.fetch
     data:
       fair_organizer_id: fairOrg.id
+      sort: "-start_at"
```
Comment (starsirius) [inline src/mobile/apps/fair_organizer/routes.coffee]: 👍to do it on the API level.

Comment (starsirius) [reply src/mobile/apps/fair_organizer/routes.coffee]: One thing I would note is that from the comment above (on line 40), it seems to be a product decision to fetch "past" fairs (and also the articles from those past fairs on the same page.) I am not very familiar with the fairs page on mobile, so I would suggest reaching out to the Fairs team and @joanna20, and talk about the changes here (i.e. now we will fetch all the fairs by the same organizer sorted by starting date (and also their articles).

If people feel good about this change, let's also modify the comment.

**Comment by starsirius** [comment]
Alright, going to merge this. We can follow up with more improvements!

**Comment by williardx** [comment]
Thanks for reviewing @starsirius!

**Pull request merged**